xdsl/dialects/experimental/aie.py

```diff
@@ -0,0 +1,1129 @@
+    packet_type = attr_def(IntegerAttr[IntegerType])
+    packet_id = attr_def(IntegerAttr[IntegerType])
+
+    def __init__(
+        self, packet_type: IntegerAttr[IntegerType], packet_id: IntegerAttr[IntegerType]
+    ):
+        super().__init__(
+            attributes={"packet_type": packet_type, "packet_id": packet_id}
+        )
+
+    def print(self, printer: Printer):
+        printer.print(
+            "(",
+            f"0x{self.packet_type.value.data:X}",
+            ", ",
+            f"0x{self.packet_id.value.data:X}",
+            ")",
+        )
+
+
+# TODO: add successor basic blocks
+@irdl_op_definition
+class DMAStartOp(IRDLOperation):
+    name = "AIE.dmaStart"
+    channelDir = attr_def(IntegerAttr[Annotated[IntegerType, i32]])
+    channelIndex = attr_def(IntegerAttr[IntegerType])
+    dest = successor_def()
+    chain = successor_def()
+
+    traits = frozenset([IsTerminator()])
+
+    def __init__(
+        self,
+        channelDir: IntegerAttr[IntegerType],
+        channelIndex: IntegerAttr[IntegerType],
+        dest: Block,
+        chain: Block,
+    ):
+        super().__init__(
+            attributes={"channelDir": channelDir, "channelIndex": channelIndex},
+            successors=[dest, chain],
+        )
+
+    def print(self, printer: Printer):
+        direction = "MM2S" if self.channelDir.value.data == 0 else "S2MM"
+        printer.print("(", direction, ", ", self.channelIndex.value.data, ", ")
+        printer.print_block_name(self.dest)
+        printer.print(", ")
+        printer.print_block_name(self.chain)
+        printer.print(")")
+
+
+@irdl_op_definition
+class DebugOp(IRDLOperation):
+    name = "AIE.debug"
+    arg = operand_def(AnyAttr())
+
+    def __init__(self, arg: Operation | SSAValue):
+        super().__init__(operands=[arg])
+
+
+@irdl_op_definition
+class DeviceOp(IRDLOperation):
+    name = "AIE.device"
+
+    region = region_def()
+
+    device = attr_def(IntegerAttr[Annotated[IntegerType, i32]])
+    traits = frozenset([SymbolTable()])
+
+    def __init__(self, device: IntegerAttr[IntegerType], region: Region):
+        super().__init__(attributes={"device": device}, regions=[region])
+
+    def print(self, printer: Printer):
+        printer.print("(")
+        device_str = "xcvc1902" if self.device.value.data == 0 else ""
+        printer.print(device_str)
+        printer.print(") ")
+        printer.print_region(self.region)
+
+
+@irdl_op_definition
+class ExternalBufferOp(IRDLOperation):
+    name = "AIE.external_buffer"
+
+    sym_name = attr_def(StringAttr)
+    buffer = result_def(memref.MemRefType)
+
+    def __init__(
+        self,
+        sym_name: str,
+        shape: ArrayAttr[AnyIntegerAttr],
+        element_type: Attribute,
+    ):
+        super().__init__(
+            attributes={"sym_name": StringAttr(sym_name)},
+            result_types=[
+                memref.MemRefType.from_element_type_and_shape(element_type, shape)
+            ],
+        )
```

Comment:
I would recommend keeping the inits as simple as possible, and letting the client deal with constructing the values.
```suggestion
    def __init__(
        self,
        sym_name: str,
        buffer: memref.MemRefType[Attribute],
    ):
        super().__init__(
            attributes={"sym_name": StringAttr(sym_name)},
            result_types=[buffer],
        )
```